clang/lib/CodeGen/CGStmt.cpp

```diff
@@ -1109,6 +1124,10 @@ void CodeGenFunction::EmitForStmt(const ForStmt &S,
   llvm::BasicBlock *CondBlock = CondDest.getBlock();
   EmitBlock(CondBlock);
 
+  if (getTarget().getTriple().isSPIRVLogical())
```

Comment:
None of these target checks have anything to do with the target. If you want to conditionally enable these for bringup, these should be hidden in an enableConvergenceTokenEmission or something predicate

Reply:
Right, something at the module level, which hides this into a more genering function, like the `shouldEmitRTTI`. Will fix.

Reply:
Changed those for a module-level function which hides the target specific bit behind a more generic check.
Let me know if that's not what you had in mind!